### PR TITLE
feature: Enable WASM on Blazor

### PR DESCRIPTION
**Pull request by richbryant**
**What kind of change does this PR introduce?**

Bug fix.  Changes to Registrations class to enable ReactiveUI.Blazor to determine operating platform effectively, Scheduler initialization

**What is the current behavior?**

mono-wasm throws errors when initializing ReactiveCommand

**What is the new behavior?**

No errors, commands work.

**What might this PR break?**

Potential to break ReactiveUI.Blazor on Server *IF* scheduling changes.  Potential to become obsolete once mono-wasm utilizes wasm threading.

**Other information**:

Requires latest PR raised against System.Reactive.Wasm

### CODE REVIEW

File: src/ReactiveUI.Blazor/ReactiveUI.Blazor.csproj [hunk 1]
```diff
@@ -7,11 +7,11 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Reactive.Wasm" Version="1.0.1-preview.12" />
     <PackageReference Include="Microsoft.AspNetCore.Components" Version="3.1.0" />
   </ItemGroup>
 
   <ItemGroup>
+    <ProjectReference Include="..\..\..\..\Users\richardb\source\repos\richbryant\Reactive.Wasm\src\System.Reactive.Wasm\System.Reactive.Wasm.csproj" />
```
Comment (glennawatson) [inline src/ReactiveUI.Blazor/ReactiveUI.Blazor.csproj]: This is likely to be an issue. I would recommend just doing a NuGet reference.

Comment (richbryant) [reply src/ReactiveUI.Blazor/ReactiveUI.Blazor.csproj]: on it

Comment (glennawatson) [reply src/ReactiveUI.Blazor/ReactiveUI.Blazor.csproj]: You can use NuGet wildcard versioning. Eg 1.* and then we can set the version of system.reactive.wasm to match.

Comment (richbryant) [reply src/ReactiveUI.Blazor/ReactiveUI.Blazor.csproj]: Nuts.  Okay. Will commit again.

Comment (richbryant) [reply src/ReactiveUI.Blazor/ReactiveUI.Blazor.csproj]: Build failed with wildcard versioning so I had to revert that.  Please feel free to amend this if you can make it build :(

File: src/ReactiveUI.Blazor/Registrations.cs [hunk 2]
```diff
@@ -27,18 +25,13 @@ public void Register(Action<Func<object>, Type> registerFunction)
...
-            else
-#endif
-            {
-                RxApp.TaskpoolScheduler = TaskPoolScheduler.Default;
-                RxApp.MainThreadScheduler = new WaitForDispatcherScheduler(() => CurrentThreadScheduler.Instance);
+                PlatformEnlightenmentProvider.Current.EnableWasm();
             }
+
+            RxApp.TaskpoolScheduler = TaskPoolScheduler.Default;
+            RxApp.MainThreadScheduler = new WaitForDispatcherScheduler(() => CurrentThreadScheduler.Instance);
```
Comment (anaisbetts) [inline src/ReactiveUI.Blazor/Registrations.cs]: We should never have to wait for the dispatcher to get CurrentThread, there is no dispatcher!

Comment (richbryant) [reply src/ReactiveUI.Blazor/Registrations.cs]: I tried to change as little as possible but I can remove that line and test again if you like.

File: src/ReactiveUI.Blazor/ReactiveUI.Blazor.csproj (hunk 1 above)
Comment (glennawatson) [reply src/ReactiveUI.Blazor/ReactiveUI.Blazor.csproj]: You should be able to wildcard now that I released 1.2.1

Comment (richbryant) [reply src/ReactiveUI.Blazor/ReactiveUI.Blazor.csproj]: it's 01:13Hrs here.  I'll do that tomorrow :) Probably mistake-prone right now.

File: src/ReactiveUI.Blazor/ReactiveUI.Blazor.csproj [hunk 3]
```diff
@@ -9,10 +9,7 @@
   <ItemGroup>
     <PackageReference Include="Reactive.Wasm" Version="1.*" />
     <PackageReference Include="Microsoft.AspNetCore.Components" Version="3.1.0" />
-  </ItemGroup>
-
-  <ItemGroup>
-    <ProjectReference Include="..\ReactiveUI\ReactiveUI.csproj" />
```
Comment (glennawatson) [inline src/ReactiveUI.Blazor/ReactiveUI.Blazor.csproj]: It should be a ProjectReference for the ReactiveUI.Csproj file.

File: src/ReactiveUI.Blazor/Registrations.cs (hunk 2 above)
Comment (glennawatson) [reply src/ReactiveUI.Blazor/Registrations.cs]: The name of `WaitForDispatcherScheduler` is deceptive. It just checks for an exception, if there is one just returns the `CurrentThreadScheduler` instead.

Comment (glennawatson) [reply src/ReactiveUI.Blazor/Registrations.cs]: So it has no reliance on the dispatcher directly. It was intended to be used with the DispatcherScheduler though.

**Pull request merged**